llvm/include/llvm/CodeGen/MachineFrameInfo.h

```diff
@@ -180,14 +180,19 @@ class MachineFrameInfo {
     /// If true, the object has been sign-extended.
     bool isSExt = false;
 
+    /// If true we can guarantee a reference to this object does not escape the
+    /// function
+    bool isSafe;
```

Comment:
I worry that the safe/unsafe naming in this commit might be a bit generic, maybe needs something like a "temporally" prefix? Also makes it a lot more obvious which code is upstream and which code isn't without staring at git blame.